src/dotnet/App.Maui/Platforms/Android/AndroidNotificationPermissions.cs

```diff
@@ -0,0 +1,54 @@
+using ActualChat.Notification.UI.Blazor;
+using Android;
+using Android.App;
+using Android.Content.PM;
+using AndroidX.Core.Content;
+
+namespace ActualChat.App.Maui;
+
+public class AndroidNotificationPermissions : INotificationPermissions
+{
+    public AndroidNotificationPermissions()
+    { }
+
+    public Task<PermissionState> GetNotificationPermissionState(CancellationToken cancellationToken)
+    {
+        var activity = Platform.CurrentActivity!;
+        if (ContextCompat.CheckSelfPermission(activity, Manifest.Permission.PostNotifications) == Permission.Granted)
+            return Task.FromResult(PermissionState.Granted);
+        return Task.FromResult(activity.ShouldShowRequestPermissionRationale(Manifest.Permission.PostNotifications)
+            ? PermissionState.Denied
+            : PermissionState.Prompt);
+    }
+
+    public async Task RequestNotificationPermissions(CancellationToken cancellationToken)
+    {
+        var activity = Platform.CurrentActivity!;
+        var state = await GetNotificationPermissionState(cancellationToken);
+
+        if (state == PermissionState.Granted)
+            return;
+
+        if (state == PermissionState.Denied) {
+            new AlertDialog.Builder(activity)
+                .SetTitle("Enable app permissions")!
+                .SetMessage("Grant notifications permission to receive push notifications for new messages")!
+                .SetNegativeButton("No thanks", (_, _) => { })!
+                .SetPositiveButton("Continue",
+                    (_, _) =>  RequestPermission())!
+                .Show();
```

Comment:
And btw, since it's not the first time I make such edits:
- This specific request is crucial, i.e. a lot depends on how user reacts to it. So it makes sense to invest 10X into phrasing it well.
- IMO you are experienced enough to draw such conclusions yourself.
- So please create Google Drive document to phrase such statements, check spelling & grammar. Please do some research on how similar requests look in other apps.